base/precompilation.jl

```diff
@@ -817,7 +825,16 @@ function _precompilepkgs(pkgs::Vector{String},
                             t = @elapsed ret = precompile_pkgs_maybe_cachefile_lock(io, print_lock, fancyprint, pkg_config, pkgspidlocked, hascolor) do
                                 Base.with_logger(Base.NullLogger()) do
                                     # The false here means we ignore loaded modules, so precompile for a fresh session
-                                    Base.compilecache(pkg, sourcepath, std_pipe, std_pipe, false; flags, cacheflags, isext = haskey(exts, pkg))
+                                    keep_loaded_modules = false
+                                    if haskey(exts, pkg)
+                                        # any extension in our direct dependencies is one we have a right to load
+                                        loadable_exts = filter((dep)->haskey(exts, dep), depsmap[pkg])
```

Comment:
The `loadable_extensions` has to be computed from the loading code, no? There is no guarantee that it is the parallel precompile code that precompiles the extension?

Reply:
Hmm, I think you're right - though it sounds like it should have been computing `isext` already (to avoid cycles) and was not.